MapboxNavigation/CarPayCompassView.swift

```diff
@@ -0,0 +1,68 @@
+import UIKit
+import Mapbox
+import Turf
+import MapboxCoreNavigation
+
+@objc(MBCarPlayCompassView)
+open class CarPlayCompassView: StylableView {
+    
+    @objc weak open var label: StylableLabel!
+    
+    // Round to closest 45° to only show main winds ["N", "NE", "E", "SE", "S", "SW", "W", "NW"]
+    static let granularity: CLLocationDirection = 360 / 8
+    
+    lazy var formatter: MGLCompassDirectionFormatter = {
+        let formatter = MGLCompassDirectionFormatter()
+        formatter.unitStyle = .short
+        return formatter
+    }()
+    
+    @objc
+    open var course: CLLocationDirection = 0 {
+        didSet {
+            if course >= 0 {
+                snappedCourse = course.wrap(min: 0, max: 360)
+            }
+        }
+    }
+    
+    fileprivate var _snappedCourse: CLLocationDirection = 0
+    fileprivate var snappedCourse: CLLocationDirection {
+        set {
+            let snappedCourse = CarPlayCompassView.granularity * round(newValue / CarPlayCompassView.granularity)
+            _snappedCourse = snappedCourse
+            label.text = formatter.string(fromDirection: snappedCourse).uppercased()
+        }
+        get {
+            return _snappedCourse
+        }
+    }
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        commonInit()
+    }
+    
+    required public init?(coder decoder: NSCoder) {
+        super.init(coder: decoder)
+        commonInit()
+    }
+    
+    func commonInit() {
+        isHidden = true
+        let label = StylableLabel(frame: .zero)
+        label.translatesAutoresizingMaskIntoConstraints = false
+        addSubview(label)
+        self.label = label
+        
+        course = 0
+        
+        widthAnchor.constraint(equalToConstant: 30).isActive = true
```

Comment:
Is this wide enough to accommodate a four-character-long string?

https://github.com/mapbox/mapbox-gl-native/blob/5b28c1eb9c1bf58bb2ab050d0e52aa7f96e8eef8/platform/darwin/resources/uk.lproj/Foundation.strings#L128
https://github.com/mapbox/mapbox-gl-native/blob/5b28c1eb9c1bf58bb2ab050d0e52aa7f96e8eef8/platform/darwin/resources/he.lproj/Foundation.strings#L128

Maybe the label’s font size needs to shrink if it doesn’t fit.

Reply:
It was a bit out of bounds and it did truncate without font scaling, but it's fixed now.